packages/composite-checkout/src/components/checkout-step.js

```diff
@@ -86,7 +86,7 @@ function CheckoutStepHeader( {
 			<Stepper isComplete={ isComplete } isActive={ isActive }>
 				{ stepNumber }
 			</Stepper>
-			<StepTitle className="checkout-step__title" isActive={ isActive }>
+			<StepTitle className="checkout-step__title" stepNumber={ stepNumber } isActive={ isActive }>
```

Comment:
In the future it's possible some host might want to add another step before the order summary, or there might be some reason to allow the order summary step to be editable. If it's possible I'd like to come up with a way to keep the step title from having logic specific to different steps.

What if we instead used the presence of the edit button as a flag to determine the width of the title? So something like this:

```js
const shouldShowEditButton = onEdit && isComplete && ! isActive;
//...
<StepTitle className="checkout-step__title" fullWidth={ ! shouldShowEditButton } isActive={ isActive }>
//...
margin-right: ${props => ( props.fullWidth ? '0' : '8px' )};
flex: ${props => ( props.fullWidth ? '1' : 'inherit' )};
```

Reply:
I like that idea! I'll update it.